### ISSUE TITLE
Issue between `plot_style` keyword and `widget`plotting backend

### ISSUE THREAD

**Issue by amcar-dtu** [hyperspy/hyperspy#3448]
#### Describe the bug
The `plot_style` keyword of the method `plot` doesn't work anymore with the `matplotlib` `widget` backend, while it works with the `ipympl` one.

#### To Reproduce
MWE
```python
%matplotlib widget
import hyperspy.api as hs

data = np.random.random((100, 100))

s = hs.signals.Signal1D(data)
s.plot(
    colorbar=True, 
    tight_layout=True,
    scalebar='all',
    cmap='mpl_colors',
    plot_style="horizontal",
    navigator_kwds=dict(
        colorbar=True,
        scalebar_color='white',
        axes_ticks=True,
    )
)
```

#### Expected behavior
Check the figure below:
<img width="1307" alt="image" src="https://github.com/user-attachments/assets/0ae93f06-44b8-4059-9e8c-ee42699c67a0">

Error message:
```
UserWarning: The `plot_style` keyword is only used when the `ipympl` or `widget`plotting backends are used.
  warnings.warn(
```

It works well with `widget` backend:
![image](https://github.com/user-attachments/assets/bb7ff757-0391-4f6a-bc65-8e9122e71b0a)

#### Python environment:
 - HyperSpy version: 2.1.1
 - Python version: 3.12.6

#### Context-relevant Information
VSCode is used for executing the code, and as pointed out by @CSSFrancis VSCode recommends the use of `matplotlib`s `widget` plotting backend over `ipympl`

**Comment by CSSFrancis** [comment]
@amcar-dtu I think (someone correct me if I'm wrong) that the `widget` and the `ipympl` backends are the same with a preference twoards using `ipympl`.  (Edit: It's seems mostly irrelevent which are are using although certain backends only support one or the other)

I'd imagine that you are using VS Code? This might be relevent  https://github.com/matplotlib/ipympl/issues/529

**Comment by amcar-dtu** [comment]
> @amcar-dtu I think (someone correct me if I'm wrong) that the `widget` and the `ipympl` backends are the same with a preference twoards using `ipympl`. (Edit: It's seems mostly irrelevent which are are using although certain backends only support one or the other)
> 
> I'd imagine that you are using VS Code? This might be relevent [matplotlib/ipympl#529](https://github.com/matplotlib/ipympl/issues/529)

@CSSFrancis correct, indeed the warning you see in the case of the working case (so for `ipympl` plotting backend) is given by VSCode and says `'%matplotlib' widget works best inside of VS Code | Jupyter`.

Since it's a relevant point, I'll update the OP, but I'd also say that the issue here is relevant since the error message from `hyperspy` itself mentions the `widget` plotting backend, so even while using such backend one gets this error message and I guess this should not happen.

**Comment by CSSFrancis** [comment]
> @CSSFrancis correct, indeed the warning you see in the case of the working case (so for `ipympl` plotting backend) is given by VSCode and says `'%matplotlib' widget works best inside of VS Code | Jupyter`.

That seems relevant.  I'd imagine that this: https://github.com/hyperspy/hyperspy/pull/3437 should fix your issue although it would be good to check.  Maybe we can fast track a 2.1.2 release to get the bug fix in. @ericpre any ideas, this seems related but not quite the same as the issue that @jlaehne was discribing.

Edit: If you are comfortable downloading the main branch and building from a development enviorment you can check that this fix works otherwise we can try to see if there is a different root cause!

**Comment by amcar-dtu** [comment]
> > @CSSFrancis correct, indeed the warning you see in the case of the working case (so for `ipympl` plotting backend) is given by VSCode and says `'%matplotlib' widget works best inside of VS Code | Jupyter`.
> 
> That seems relevant. I'd imagine that this: #3437 should fix your issue although it would be good to check. Maybe we can fast track a 2.1.2 release to get the bug fix in. @ericpre any ideas, this seems related but not quite the same as the issue that @jlaehne was discribing.
> 
> Edit: If you are comfortable downloading the main branch and building from a development enviorment you can check that this fix works otherwise we can try to see if there is a different root cause!

@CSSFrancis seems to be the case: I added the branch referenced in [https://github.com/hyperspy/hyperspy/pull/3437](https://github.com/hyperspy/hyperspy/pull/3437) to my dev dependencies:

```toml
[tool.poetry.dev-dependencies]
hyperspy = {git = "https://github.com/hyperspy/hyperspy", branch = "RELEASE_next_patch"}
```

and that's the result!

![image](https://github.com/user-attachments/assets/bce7ad90-c736-479c-825f-df464cf58ed4)

**Comment by CSSFrancis** [comment]
@amcar-dtu Beautiful! :)

**Comment by ericpre** [comment]
> That seems relevant. I'd imagine that this: #3437 should fix your issue although it would be good to check. Maybe we can fast track a 2.1.2 release to get the bug fix in. @ericpre any ideas, this seems related but not quite the same as the issue that @jlaehne was discribing.

@CSSFrancis, I would be keen to make a minor release, there are several features that are useful in the minor branch.

**Comment by ericpre** [comment]
Closing since it is fixed in #3437.
